File: test/public/cgi-bin/hello.rb
#!/usr/bin/env ruby

require 'cgi'

cgi = CGI.new

#puts "Content-Type: text/html\n"
puts cgi.header
puts "<h2 style='text-align: center;'>Hello from Ruby to Node.js!</h2>"

#data = gets()
#if data
#  puts data
#end


fields = cgi.keys

if fields
  puts "<ul>"
  for field in fields
    puts "<li>#{field} = <strong>#{cgi[field]}</strong></li>"
  end
  puts "</ul>"
else
 puts "No CGI input"
end

puts "<a href='javascript: history.back();'>Back to Form</a>"

File: test/public/cgi-bin/hello.c
#!/usr/bin/tcc -run

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, char *argv[]) {
    printf("Content-Type: text/html\n\n");
    printf("<title>Hello C Version</title>");
    printf("<h1 style='text-align: center; color: #880000;'>Hello, C World!</h1>");
    char *qs = getenv("QUERY_STRING");
    if (qs && (strlen(qs) > 0)) {
	printf("<p>QUERY STRING: <strong>%s</strong></p>", qs);
    }
    else {
	printf("<p>No QUERY STRING ;-(</p>");
    }

    char input[256];
    scanf("%s", input);
    printf("<p>STDIN: <strong>%s</strong></p>", input);
    
    return 0;
}

